src/entity-manager/EntityManager.ts

```diff
@@ -178,7 +180,9 @@ export class EntityManager extends BaseEntityManager {
         } else {
             // todo: throw exception if constructor in target is not set
             if (target instanceof Array) {
-                return this.getRepository<Entity[]>(target[0].constructor).remove(entity as Entity[], options);
+                return Promise.all(target.map((t, i) => {
```

Comment:
I do not understand. Are you referring to `t` and `i`? If we're discussing trying to make the code more clear, let's talk about how `target` and `entity` are singular words, but here in the code, they actually both arrays (so `targets` and `entities`) and referring the _same_ array... it's as confusing as heck 😄.

I thought about trying to clear that up a bit, but I try to make as few changes as I can in PRs for large projects if I can help it.